pytest fails with non-None return

The latest version of *pytest* (8.4.0) now automatically fails a test that has a return value (see the [changelog](https://docs.pytest.org/en/stable/changelog.html#pytest-8-4-0-2025-06-02) and the [PR with the change](https://github.com/pytest-dev/pytest/pull/12346)).

From a little grepping, this change only affects one test, *test_get_component_name* in `stage_1/info_test.py`.

https://github.com/csdms/bmi-tester/blob/c4f5c994a13f6fd010ef5ba25fdcaffa893222be/src/bmi_tester/_tests/stage_1/info_test.py#L8-L13

This error popped up in a monthly test of *bmi-topography*:
```
============================= test session starts ==============================
platform darwin -- Python 3.13.3, pytest-8.4.0, pluggy-1.6.0 -- /Users/runner/work/bmi-topography/bmi-topography/.nox/test-bmi/bin/python
cachedir: .pytest_cache
rootdir: /Users/runner/work/bmi-topography/bmi-topography
configfile: pyproject.toml
plugins: dependency-0.6.0
collecting ... collected 12 items

../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/info_test.py::test_get_component_name FAILED [  8%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/info_test.py::test_var_names[land_surface__elevation] PASSED [ 16%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/info_test.py::test_input_var_name_count SKIPPED [ 25%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/info_test.py::test_output_var_name_count SKIPPED [ 33%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/info_test.py::test_get_input_var_names SKIPPED [ 41%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/info_test.py::test_get_output_var_names SKIPPED [ 50%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/time_test.py::test_get_start_time PASSED [ 58%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/time_test.py::test_get_time_step PASSED [ 66%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/time_test.py::test_time_units_is_str PASSED [ 75%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/time_test.py::test_time_units_is_valid SKIPPED [ 83%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/time_test.py::test_get_current_time SKIPPED [ 91%]
../.nox/test-bmi/lib/python3.13/site-packages/bmi_tester/_tests/stage_1/time_test.py::test_get_end_time SKIPPED [100%]😞 There were errors


=================================== FAILURES ===================================
___________________________ test_get_component_name ____________________________
Expected None, but test returned 'bmi-topography'. Did you mean to use `assert` instead of `return`?
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -28,7 +28,6 @@ jobs:
       - uses: conda-incubator/setup-miniconda@v3
         with:
           python-version: ${{ matrix.python-version }}
-          miniforge-variant: Mambaforge
           miniforge-version: latest
 
       - name: Test

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -57,9 +57,8 @@ repos:
         )
 
 - repo: https://github.com/pre-commit/mirrors-mypy
-  rev: v1.8.0
+  rev: v1.16.0
   hooks:
   - id: mypy
     language_version: python3.12
-    additional_dependencies: [types-all]
     files: bmi_tester/.*\.py$

--- a/src/bmi_tester/_tests/stage_1/info_test.py
+++ b/src/bmi_tester/_tests/stage_1/info_test.py
@@ -10,8 +10,6 @@ def test_get_component_name(initialized_bmi):
     name = initialized_bmi.get_component_name()
     assert isinstance(name, str)
 
-    return name
-
 
 def test_var_names(var_name):
     """Test var names are valid."""

